test/extended/cli/template.go

```diff
@@ -16,10 +18,299 @@ var _ = g.Describe("[sig-cli] templates", func() {
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.Equal("List"))
+		out, err = oc.Run("process").Args("ruby-helloworld-sample", "-o", "go-template={{.kind}}").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.Equal("List"))
+		err = oc.Run("process").Args("ruby-helloworld-sample", "-o", "go-template-file=/dev/null").Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		out, err = oc.Run("process").Args("ruby-helloworld-sample", "-o", "jsonpath", "--template", "{.kind}").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.Equal("List"))
+		out, err = oc.Run("process").Args("ruby-helloworld-sample", "-o", "jsonpath={.kind}").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.Equal("List"))
+		err = oc.Run("process").Args("ruby-helloworld-sample", "-o", "jsonpath-file=/dev/null").Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		out, err = oc.Run("process").Args("ruby-helloworld-sample", "-o", "describe").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("ruby-27-centos7"))
+		out, err = oc.Run("process").Args("ruby-helloworld-sample", "-o", "json").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("ruby-27-centos7"))
+		out, err = oc.Run("process").Args("ruby-helloworld-sample", "-o", "yaml").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("ruby-27-centos7"))
+		out, err = oc.Run("process").Args("ruby-helloworld-sample", "-o", "name").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("ruby-27-centos7"))
+		out, err = oc.Run("describe").Args("templates", "ruby-helloworld-sample").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("ruby-sample-build"))
+		err = oc.Run("delete").Args("-f", appTemplatePath).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		outputTemplateFile, err := oc.Run("process").Args("-f", guestbookTemplatePath, "-l", "app=guestbook").OutputToFile("guestbooktemplate.yaml")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = oc.Run("create").Args("-f", outputTemplateFile).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		out, err = oc.Run("status").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("frontend-service"))
+		out, err = oc.Run("process").Args("-f", guestbookTemplatePath, "--local", "-l", "app=guestbook", "-o", "yaml").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("app: guestbook"))
+		out, err = oc.Run("process").Args("-f", guestbookTemplatePath, "--local", "-l", "app=guestbook", "-o", "yaml", "--server", "0.0.0.0:1").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("app: guestbook"))
+		err = oc.Run("delete").Args("-f", outputTemplateFile).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
```

Comment:
```
os::cmd::expect_success 'oc process -f "${guestbook_template}" --local -l app=guestbook -o yaml ADMIN_USERNAME=au ADMIN_PASSWORD=ap REDIS_PASSWORD=rp > "${new}/localtemplate"'
os::cmd::expect_success 'oc process -f "${guestbook_template}" -l app=guestbook -o yaml ADMIN_USERNAME=au ADMIN_PASSWORD=ap REDIS_PASSWORD=rp > "${new}/remotetemplate"'
os::cmd::expect_success 'diff "${new}/localtemplate" "${new}/remotetemplate"'
```
lines were skipped. On purpose?

Reply:
First reason is I don't know we support external tools like `diff` or `cat` in origin go tests. Second and more important reason is there are several repetitive tests doing the same thing and these skipped tests are only testing the same thing by combining these external tools.

Reply:
In this case testing `--local` and non-local case produce the same output is still a test case. Worth keeping the case even without the diff. There's `CLI.InputString()` method that can emulate `cat`/`echo` cases to test the `--param-file=-` as well. 

> there are several repetitive tests doing the same thing and these skipped tests are only testing the same thing by combining these external tools

Are you referring to the `--param-file=FILE` and `--param-file=-` cases?

Reply:
> In this case testing `--local` and non-local case produce the same output is still a test case. Worth keeping the case even without the diff. There's `CLI.InputString()` method that can emulate `cat`/`echo` cases to test the `--param-file=-` as well.
> 

--local and --non-local good be a good test case. Thanks.
> > there are several repetitive tests doing the same thing and these skipped tests are only testing the same thing by combining these external tools
> 
> Are you referring to the `--param-file=FILE` and `--param-file=-` cases?

Actually I'm referring to all tests in this file.